### PR TITLE
add pipeline parameters and ability to bind them to values passed fro…

### PR DESCRIPTION
**Pull request by bolyachevets**
…m webhook

1. We need global pipeline parameters that can be set via webhook trigger and shared by the tasks - hence add pipeline_params
https://tekton.dev/docs/pipelines/pipelines/#specifying-parameters

2. We need to bind these parameters to values received from webhook

3. We need to set custom interceptors to filter webhooks

4. Maven build needs more pod resource than default setting, mounting of settings.xml, and pvc volumes (e.g., I have been using a pvc to cache artifactory repo) - hence need to be able to have more customizations for Task objects

5. Added ability to switch service accounts between tasks

6.  I started working on an example - it reacts to open/reopen pull_request gets url/commit sha from the webhook, and passes these to git-clone ClusterTask and does a maven build, then the image is build using .war generated from maven step via ocp cli ClusterTask, it is deployed in test and we subsequently run python integration tests against it with a comment being posted against the PR that contains pytest logs

### CODE REVIEW

File: charts/tekton-pipelines/templates/020-pipelines.yaml [hunk 1]
```diff
@@ -11,9 +11,15 @@ metadata:
     heritage: {{ $.Release.Service }}
 spec:
 
+  {{- if .workspaces -}}
+  {{ if gt (len .workspaces) 0 }}
   workspaces:
-    - name: shared-workspace
-    
+    {{ range .workspaces }}
```
Comment (BrandonSharratt) [inline charts/tekton-pipelines/templates/020-pipelines.yaml:17]: Would consider using .workspaces | toYaml | indent 4 here but should work fine

File: charts/tekton-pipelines/templates/500-triggerTemplate.yaml [hunk 2]
```diff
@@ -10,10 +10,10 @@ metadata:
     release: {{ $.Release.Name }}
     heritage: {{ $.Release.Service }}
 spec:
-  {{- if .params -}}
-  {{- if gt (len .params) 0 }}
+  {{- if .pipeline_params -}}
```
Comment (BrandonSharratt) [inline charts/tekton-pipelines/templates/500-triggerTemplate.yaml:13]: I suspect this needs to exist if either params or pipeline params exists and loop over both

File: charts/tekton-pipelines/templates/500-triggerTemplate.yaml [hunk 3]
```diff
@@ -22,27 +22,75 @@ spec:
...
+            taskServiceAccountName: {{ .taskServiceAccountName }}
+          {{- end -}}
+        {{- end -}}
+        {{- end }}
+
+        {{- if .pipeline_params -}}
+        {{- if gt (len .pipeline_params) 0 }}
         params:
-          {{- range .params }}
+          {{- range .pipeline_params }}
```
Comment (BrandonSharratt) [inline charts/tekton-pipelines/templates/500-triggerTemplate.yaml:87]: Same as above

File: charts/tekton-pipelines/templates/505-triggerBinding.yaml [hunk 4]
```diff
@@ -10,16 +10,16 @@ metadata:
     release: {{ $.Release.Name }}
     heritage: {{ $.Release.Service }}
 
-{{ if .params }}
-{{ if gt  (len .params) 0}}
+{{ if .pipeline_params }}
```
Comment (BrandonSharratt) [inline charts/tekton-pipelines/templates/505-triggerBinding.yaml:13]: This likely needs to consider params as well

**Review by BrandonSharratt — CHANGES_REQUESTED** [review]
See the change requests for pipeline params and params

File: charts/tekton-pipelines/templates/500-triggerTemplate.yaml (hunk 2 above)
Comment (bolyachevets) [reply charts/tekton-pipelines/templates/500-triggerTemplate.yaml:13]: I reread https://tekton.dev/docs/pipelines/pipelines/#specifying-parameters

> You can specify global parameters, such as compilation flags or artifact names, that you want to supply to the Pipeline at execution time. Parameters are passed to the Pipeline from its corresponding PipelineRun and can replace template values specified within each Task in the Pipeline.

Here is another source of inspiration: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.5/html-single/pipelines/index#about-pipelines_understanding-openshift-pipelines

Check the title for item (6):

> Parameters used across all the Tasks in the Pipeline.

and here is my interpretation (let me know if you agree/disagree) - "global pipeline parameters" (i.e. pipeline_params), which in Tekton are listed above tasks, and task parameters unfortunately are labeled as "params" in Tekton. My impression is that when creating the PipelineRun we only supply values for the "global pipeline parameters". I have almost convinced myself of this. If it was not the design, it still makes sense to separate task parameters from pipeline parameters for these reasons: 1) have a single location in the script where parameters are set 2) looping over both task and pipeline parameters might result in assigning values to the same parameter more than once

File: charts/tekton-pipelines/templates/020-pipelines.yaml (hunk 1 above)
Comment (bolyachevets) [reply charts/tekton-pipelines/templates/020-pipelines.yaml:17]: It looks like .workspaces contain some extra fields that are not used in this template, but are used in TriggerTemplate to select storage for workspaces:

    workspaces:
      - name: shared-workspace
        volumeClaimTemplate: 2Gi
      - name: input
      - name: pr-content
        volumeClaimTemplate: 50Mi

see lines 35-63 in triggerTemplate.yaml

**Pull request merged**